apps/srp.c

```diff
@@ -151,14 +153,16 @@ static char *srp_create_user(char *user, char **srp_verifier,
                              char **srp_usersalt, char *g, char *N,
                              char *passout, int verbose)
 {
-    char password[1024];
+    char password[1025];
     PW_CB_DATA cb_tmp;
     char *gNid = NULL;
     char *salt = NULL;
+    int len;
     cb_tmp.prompt_info = user;
     cb_tmp.password = passout;
 
-    if (password_callback(password, sizeof(password), 1, &cb_tmp) > 0) {
+    if ((len = password_callback(password, sizeof(password)-1, 1, &cb_tmp)) > 0) {
+        password[len] = 0;
```

Comment:
On related note there are formatting and styling problems even here. For example few lines below it should rather be
```
        if (!(gNid = SRP_create_verifier(user, password, &salt, srp_verifier,
                                         N, g)))
            BIO_printf(bio_err, "Internal error creating SRP verifier\n");
        else
            *srp_usersalt = salt;
```
It would be also appropriate to add call to OPENSSL_cleanse(passwd,len).